travel-event-host-app/src/components/searchSection/searchSection.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+import styles from './styles.module.css';
+
+import FilterBox from './searchEventsFilterBox/filterBox';
+import { useState, useEffect } from 'react';
+import { Box, MenuItem, Select, Typography } from '@mui/material';
+import EventCard from '../event/event-card/Event-card';
+import Event from '@/models/event';
+import CategoriesType from '@/components/searchSection/types';
+import { useRouter } from 'next/navigation';
+import { SearchInput } from './searchInput/searchInput';
+
+export default function SearchSection({ keyword }: { keyword: string }) {
+  const [sortBy, setSortBy] = useState<string>('Date');
+  const [resultEventList, setResultEventList] = useState<Event[]>([]);
+  const [categories, setCategories] = useState<CategoriesType>({});
+  const [filterBoxIsOpen, setFilterBoxIsOpen] = useState<boolean>(false);
+  const router = useRouter();
+
+  const handleSearch = (searchInput: string) => {
+    const url = `/searchevent/${searchInput}`; // Construct the URL
+    router.push(url); // Navigate to the URL
+  };
+
+  useEffect(() => {
+    const fetchEvents = async () => {
+      try {
+        let queryParams = '';
+
+        Object.entries(categories).forEach(([categoryName, value]) => {
+          if (value) {
+            queryParams += `&Category=${categoryName}`;
+          }
+        });
+
+        if (keyword) {
+          queryParams += `&keyword=${keyword}`;
+        }
+
+        if (queryParams.length > 0) {
+          queryParams = queryParams.substring(1);
+        }
+
+        const apiUrl = `/api/event?${queryParams}`;
+
+        const response = await fetch(apiUrl);
+        if (response.ok) {
+          const data = await response.json();
+          setResultEventList(data.events);
+          console.log(data.events);
+        } else {
+          setResultEventList([]);
+          console.error('Error', response.statusText);
+        }
+      } catch (error) {
+        console.error('Error:', error);
+      }
```

Comment:
Could we move the fetch logic into a client? `client/event/event-client`. We can create a function something like `getEventsBySearchQuery`  that returns Promise<Event[]>